packages/dara-components/dara/components/graphs/definitions.py

```diff
@@ -31,6 +32,119 @@ class EditorMode(str, Enum):
     """Resolver mode - allows users to accept edges"""
 
 
+class ArrowType(str, Enum):
+    NONE = 'none'
+    NORMAL = 'normal'
+    FILLED = 'filled'
+    EMPTY = 'empty'
+    SOFT = 'soft'
+
+
+class CenterSymbol(str, Enum):
+    NONE = 'none'
+    CROSS = 'cross'
+    QUESTION = 'question'
+    BIDIRECTED = 'bidirected'
+
+
+class Legend(BaseModel):
+    type: str
+
+    Edge: ClassVar[Type['EdgeLegend']]
+    Spacer: ClassVar[Type['SpacerLegend']]
+    Node: ClassVar[Type['NodeLegend']]
+
+
+class SpacerLegend(Legend):
+    """
+    Defines a spacer legend for a graph element.
+
+    :param label: Optional label to show in the legend
+    """
+
+    type: Literal['spacer'] = 'spacer'
```

Comment:
The type properties should not be modifiable. To achieve that in pydantic you can do 
```suggestion
    type: Literal['spacer'] = Field(default='spacer', const=True)
```